dts/bindings/gpio/nxp,kinetis-gpio-cluster.yaml

```diff
@@ -0,0 +1,9 @@
+description: A group of GPIOs that share an interrupt.
+
+compatible: "nxp,kinetis-gpio-cluster"
```

Comment:
let's actually just call it nxp,gpio-cluster, then we can just re-use this compatible for rt595 or any other nxp platforms having the same issue when we fix those

I have ideas for a more zephyr generic solution for this type of problem but it's outside scope of this PR